txpool/txpool_test.go

```diff
@@ -4042,63 +4229,139 @@ func getDefaultEnabledForks() *chain.Forks {
 func BenchmarkAddTxTime(b *testing.B) {
 	b.Run("benchmark add one tx", func(b *testing.B) {
 		signer := crypto.NewEIP155Signer(100)
-
 		key, err := crypto.GenerateECDSAPrivateKey()
-		if err != nil {
-			b.Fatal(err)
-		}
+		require.NoError(b, err)
 
 		signedTx, err := signer.SignTx(newTx(crypto.PubKeyToAddress(&key.PublicKey), 0, 1, types.LegacyTxType), key)
-		if err != nil {
-			b.Fatal(err)
-		}
+		require.NoError(b, err)
 
 		for i := 0; i < b.N; i++ {
 			pool, err := newTestPool()
-			if err != nil {
-				b.Fatal("fail to create pool", "err", err)
-			}
-
+			require.NoError(b, err, "fail to create pool")
 			pool.SetSigner(signer)
 
 			err = pool.addTx(local, signedTx)
-			if err != nil {
-				b.Fatal(err)
-			}
+			require.NoError(b, err)
 		}
 	})
 
 	b.Run("benchmark fill account", func(b *testing.B) {
 		signer := crypto.NewEIP155Signer(100)
-
 		key, err := crypto.GenerateECDSAPrivateKey()
-		if err != nil {
-			b.Fatal(err)
-		}
+		require.NoError(b, err)
 
 		addr := crypto.PubKeyToAddress(&key.PublicKey)
-		txs := make([]*types.Transaction, defaultMaxAccountEnqueued)
 
+		txs := make([]*types.Transaction, defaultMaxAccountEnqueued)
 		for i := range txs {
 			txs[i], err = signer.SignTx(newTx(addr, uint64(i), uint64(1), types.LegacyTxType), key)
-			if err != nil {
-				b.Fatal(err)
+			require.NoError(b, err)
+		}
+
+		for i := 0; i < b.N; i++ {
+			pool, err := newTestPool()
+			require.NoError(b, err, "fail to create pool")
+			pool.SetSigner(signer)
+
+			for i := 0; i < len(txs); i++ {
+				err = pool.addTx(local, txs[i])
+				require.NoError(b, err)
+			}
+		}
+	})
+
+	b.Run("benchmark more accounts add one tx", func(b *testing.B) {
+		signer := crypto.NewEIP155Signer(100)
+
+		txs := make([]*types.Transaction, defaultMaxAccountEnqueued)
+		for i := range txs {
+			key, err := crypto.GenerateECDSAPrivateKey()
+			require.NoError(b, err)
+
+			addr := crypto.PubKeyToAddress(&key.PublicKey)
+
+			txs[i], err = signer.SignTx(newTx(addr, uint64(0), uint64(1), types.LegacyTxType), key)
+			require.NoError(b, err)
+		}
+
+		for i := 0; i < b.N; i++ {
+			pool, err := newTestPool()
+			require.NoError(b, err, "fail to create pool")
+			pool.SetSigner(signer)
+
+			for i := 0; i < len(txs); i++ {
+				err = pool.addTx(local, txs[i])
+				require.NoError(b, err)
+			}
+		}
+	})
+
+	b.Run("benchmark more accounts add more transactions", func(b *testing.B) {
+		signer := crypto.NewEIP155Signer(100)
+		txs := make([]*types.Transaction, accountNumber*defaultMaxAccountEnqueued)
+
+		n := (uint64)(0)
+		for n < accountNumber {
+			key, err := crypto.GenerateECDSAPrivateKey()
+			require.NoError(b, err)
+
+			addr := crypto.PubKeyToAddress(&key.PublicKey)
+
+			i := (uint64)(0)
+			for i < defaultMaxAccountEnqueued {
+				txs[n*defaultMaxAccountEnqueued+i], err = signer.SignTx(newTx(addr, i, uint64(1), types.LegacyTxType), key)
+				require.NoError(b, err)
+
+				i++
 			}
+
+			n++
 		}
```

Comment:
Suggestion to simplify the loops:
```suggestion
		for n := uint64(0); n < accountNumber; n++ {
			key, err := crypto.GenerateECDSAPrivateKey()
			require.NoError(b, err)

			addr := crypto.PubKeyToAddress(&key.PublicKey)

			for i := uint64(0); i < defaultMaxAccountEnqueued; i++ {
				txs[n*defaultMaxAccountEnqueued+i], err = signer.SignTx(newTx(addr, i, uint64(1), types.LegacyTxType), key)
				require.NoError(b, err)
			}
		}
```